packit_service/worker/checker/copr.py

```diff
@@ -138,3 +141,44 @@ def _pre_check(self) -> bool:
                 )
                 return False
         return True
+
+
+class AreFilesChanged(Checker, GetCoprBuildJobHelperForIdMixin):
+    """
+    Check if any files under the current package's `paths` field is changed.
+    If not, then just skip the current copr build job.
+    """
+
+    def get_files_changed(self) -> list[Path]:
+        """
+        Get the list of files changed in the current commit or the current pullrequest
+        """
+        # Get the changes object
+        if self.job_config.trigger == JobConfigTriggerType.pull_request:
+            changes = self.copr_build_helper.pull_request_object.changes
+        elif self.job_config.trigger == JobConfigTriggerType.commit:
+            branch = self.job_config.branch or self.project.default_branch
+            changes = self.copr_build_helper.project.get_commit(branch)
```

Comment:
Something to note here. If a PR is merged as a rebase, only the top commit will be checked for changes

Reply:
that's a good point. Could we get and check all the commits?

Reply:
:thinking: Maybe it's possible. At least with Pagure we know that there is a link between commit and PR, and maybe there is one for Github as well. Is there a PR merged as rebase (didn't find better wording) in https://github.com/packit/hello-world ?

Reply:
I think we could use the `commits` directly from the webhook payload [here](https://docs.github.com/en/webhooks/webhook-events-and-payloads#push), wdyt? It would require changing the [parsing](https://github.com/packit/packit-service/blob/cdce0c7cc1b33c28b07fd87c494d6956ff471328/packit_service/worker/parser.py#L521)  and extending the [class](https://github.com/packit/packit-service/blob/main/packit_service/events/github/push.py). Then in the checker we would check this based on the event type.

Reply:
same can also be done for [Gitlab](https://docs.gitlab.com/user/project/integrations/webhook_events/#push-events). Another benefit is having the information directly, without additional API calls.

> We do have a hook we can use there:

is this deterministic though? (e.g. one commit associated to multiple PRs)